src/main/scala/pirate/Parser.scala

```diff
@@ -4,21 +4,26 @@ import scalaz._
 
 sealed trait Parser[A] {
   def map[B](f: A => B): Parser[B] = this match {
-    case SwitchParser(flag, a) =>
-      SwitchParser(flag, f(a))
-    case FlagParser(flag, metas, p) =>
-      FlagParser(flag, metas, p.map(f))
-    case ArgumentParser(p) =>
-      ArgumentParser(p.map(f))
-    case CommandParser(name, p) =>
-      CommandParser(name, p.map(f))
+    case SwitchParser(meta, a) =>
+      SwitchParser(meta, f(a))
+    case FlagParser(meta, p) =>
+      FlagParser(meta, p.map(f))
+    case ArgumentParser(meta, p) =>
+      ArgumentParser(meta, p.map(f))
+    case CommandParser(sub) =>
+      CommandParser(sub.copy(parse = sub.parse.map(f)))
+  }
+
+  def isArg: Boolean = this match {
+    case ArgumentParser(_, _) => true
+    case _                    => false
   }
 }
 
-case class SwitchParser[A](flag: Name, a: A) extends Parser[A]
-case class FlagParser[A](flag: Name, metas: List[String], p: Read[A]) extends Parser[A]
-case class ArgumentParser[A](p: Read[A]) extends Parser[A]
-case class CommandParser[A](name: String, p: Parse[A]) extends Parser[A]
+case class SwitchParser[A](meta: Metadata, a: A) extends Parser[A]
```

Comment:
Is there any reason why the Parser needs all the Metadata, or is it just a convenience?